### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by binodbhusal**
The following tasks were carried out in this milestone

- [x] Removed the current_user method in ApplicationController, devise will provide us with one.

- [x] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).

- [x] Can register a new user.
- [x] User logs in with a combination of email and password.
- [x] Hashed password should be stored in the database.
- [x] Ask for confirmation of email.
- [x] Can reset password.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @binodbhusal,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly collaborate with your coding partner.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (ridwanediallo) [inline app/models/user.rb]: - [ ] Well done adding the devise to your project. However, there is one requirement with devise that is missing with is to ensure that the user should confirm their email before being able to sign up on the app.

### Requirement :point_down: 
Screenshot from 2023-09-07 14-50-55


So to achieve that requirement you should enable the `:confirmable` feature. Please do as follow :point_down: 
```suggestion
  # Include default devise modules. Others available are:
  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
  devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable, :confirmable
```

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up", class:"btn-sign"  %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
+</div>
```
Comment (ridwanediallo) [inline app/views/devise/registrations/new.html.erb]: - [ ] Please make sure the sign-up page's design adheres to the wireframe. You can completely customize your visual design based on the requirements, but you should maintain the wireframe. 👍️

### wireframe :point_down: 

Screenshot from 2023-09-07 15-07-12

### Note: :point_down: 
This wireframe, as you can see, is for the sign-in page, but please make sure to adapt the same design for the sign-up page. The sign-up form contains more inputs than the others, but other than that, they should have a similar layout. :ok_hand: 😊

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in", class:"btn-sign" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
+</div>
```
Comment (ridwanediallo) [inline app/views/devise/sessions/new.html.erb]: - [ ] Please make sure the sign-in page's design adheres to the wireframe. You can completely customize your visual design based on the requirements, but you should maintain the wireframe. 👌 😊

### wireframe :point_down: 

Screenshot from 2023-09-07 15-07-12

**Review by ridwanediallo — APPROVED** [review]
# Changes Required ♻️
## Hi :wave @binodbhusal and @ybadr99 , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow is used correctly. ✔
- Awesome job, team! You've successfully configured devise. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi team, 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Devise comfirmable has successfully been enable. 🥳
- The sign in and sign up pages are layed out based on the wireframe. 💪
- Awesome job woking on the requested changes. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**